pkg/client/cache/store.go

```diff
@@ -76,6 +81,57 @@ func (c *cache) Add(obj interface{}) error {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 	c.items[key] = obj
+	c.updateIndices(obj)
```

Comment:
I don't think rebuilding the entire index is going to scale very long.  Why not start with incremental?  Get prev object, gen index entries for it, delete those, gen index entries for new object, insert those.


> On Feb 6, 2015, at 11:08 AM, David Eads <notifications@github.com> wrote:
> 
> In pkg/client/cache/store.go:
> 
> > @@ -76,6 +81,57 @@ func (c *cache) Add(obj interface{}) error {
> >  	c.lock.Lock()
> >  	defer c.lock.Unlock()
> >  	c.items[key] = obj
> > +	c.updateIndices(obj)
> Can you build the list of what you need to modify in the indexes separately from actually making the update? It would help limit contention on the lock. Same reason the key is calculated before lock acquisition.
> 
> —
> Reply to this email directly or view it on GitHub.
>